### ISSUE TITLE
Parachain network stale at random block after upgrading to v0.9.12

### ISSUE THREAD

**Issue by h4x3rotab** [paritytech/substrate#10158]
We found around once per day, all the Khala Network nodes can get stuck at some blocks with errors including `Storage root must match that calculated` or `Transaction will be valid in the future`.

We have seen similar problems in our early parachain testnet when the substrate dependency version of the runtime differs from that of the node client. In this time, the problems started from the upgrade of v0.9.12, while our runtime code still depends on v0.9.10.

Details here: https://github.com/Phala-Network/phala-blockchain/issues/551

**Comment by bkchr** [comment]
Caching issue, duplicate of https://github.com/paritytech/cumulus/issues/607
